pkg/genericNotes/GenericNoteHistoryService_test.go

```diff
@@ -0,0 +1,62 @@
+package genericNotes
+
+import (
+	"errors"
+	"github.com/devtron-labs/devtron/internal/util"
+	"github.com/devtron-labs/devtron/pkg/genericNotes/repository/mocks"
+	"github.com/go-pg/pg"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"testing"
+	"time"
+)
+
+func TestSave(t *testing.T) {
+
+	testInput := &GenericNoteHistoryBean{
+		NoteId:      1,
+		Description: "test description",
+		CreatedBy:   1,
+		CreatedOn:   time.Now(),
+	}
+	testUserId := int32(1)
+
+	t.Run("Test save successfully", func(tt *testing.T) {
+		noteHistoryService, mockNoteHistoryRepo := InitNoteHistoryService(t)
+		tx := &pg.Tx{}
+		mockNoteHistoryRepo.On("SaveHistory", tx, mock.AnythingOfType("*repository.GenericNoteHistory")).Return(nil)
+		resp, err := noteHistoryService.Save(tx, testInput, testUserId)
+		assert.Nil(tt, err)
+		assert.NotNil(tt, resp)
+		if resp == nil {
```

Comment:
use `assert.NotEmpty` in which u can also pass your failure msg.

Reply:
done